.github/workflows/guestconfiguration-test-run.yml

```diff
@@ -0,0 +1,86 @@
+name: Guest Configuration Test Run
+
+on:
+  workflow_call:
+    inputs:
+      target:
+        required: true
+        type: string
+      arch:
+        required: true
+        type: string
+      package-type:
+        required: true
+        type: string
+      tag:
+        required: false
+        description: 'Pester tag to include in the test run. If not provided, all tests will be run.'
+        default: ''
+        type: string
+
+jobs:
+  package:
+    uses: ./.github/workflows/package-build.yml
+    with:
+      target: ${{ inputs.target }}
+      arch: ${{ inputs.arch }}
+      artifact: ${{ inputs.target }}
+      package-type: ${{ inputs.package-type }}
+      test: true
+
+  test:
+    needs: package
+    runs-on: [self-hosted, 1ES.Pool=e2e-pool, '1ES.ImageOverride=${{ inputs.target }}']
```

Comment:
Does this absolutely need to run on a 1ES hosted runner? We should try to avoid using these runners in CI (or create a _CI-only_ pool with no access to secrets/internal resources).

Reply:
If we do need the 1ES pool for a CI job, we may also want to consider changing the pool configuration to keep a _hot_ runner (so that we do not need to wait on the pool to spin up a new runner which can take a several minutes).

Reply:
yes, we do need to have them on a hosted runners, it requires a specific VM environment. In all my tests this all happened quickly though (<2 minutes) and did not impact the end runtime as CI took longer than these tests.